src/tests/ftest/util/results_utils.py

```diff
@@ -0,0 +1,331 @@
+        """Get the value of the attribute name.
+
+        Args:
+            name (str): name of the class attribute to get
+            default (object, optional): value to return if name is not defined. Defaults to None.
+
+        Returns:
+            object: the attribute value or default if not defined
+
+        """
+        return self.get(name, default)
+
+    def __contains__(self, item):
+        """Determine if the attribute exists in this class, e.g. 'item' in self.
+
+        This is used when creating the html test result.
+
+        Args:
+            item (str): attribute name
+
+        Returns:
+            bool: True if the attribute name has a value in this object; False otherwise
+
+        """
+        return self.get(item, None) is not None
+
+    def get(self, name, default=None):
+        """Get the value of the attribute name.
+
+        Args:
+            name (str): name of the class attribute to get
+            default (object, optional): value to return if name is not defined. Defaults to None.
+
+        Returns:
+            object: the attribute value or default if not defined
+
+        """
+        try:
+            return getattr(self, name, default)
+        except TypeError:
+            return default
+
+    def start(self):
+        """Mark the start of the test."""
+        if self.time_start == -1:
+            self.time_start = time.time()
+
+    def end(self):
+        """Mark the end of the test."""
+        self.time_end = time.time()
+        self.time_elapsed = self.time_end - self.time_start
+
+
+class Results():
+    # pylint: disable=too-few-public-methods
+    """Provides the necessary result data to generate a xml/html results file."""
+
+    def __init__(self, log_file):
+        """Initialize a Results object.
+
+        Args:
+            log_file (str): the log file location for all of the tests
+        """
+        self.logfile = log_file
+        self.tests = []
+
+    @property
+    def tests_total(self):
+        """Get the total number of tests.
+
+        Returns:
+            int: total number of tests
+        """
+        return len(self.tests)
+
+    @property
+    def tests_total_time(self):
+        """Get the total test time.
+
+        Returns:
+            int: total duration of all the tests
+        """
+        return sum(test.time_elapsed for test in self.tests)
+
+    @property
+    def passed(self):
+        """Get the total number of passed tests.
+
+        Returns:
+            int: total number of passed tests
+        """
+        test_status = [str(test.status) for test in self.tests]
+        return test_status.count(TestResult.PASS)
+
+    @property
+    def warned(self):
+        """Get the total number of warned tests.
+
+        Returns:
+            int: total number of warned tests
+        """
+        test_status = [str(test.status) for test in self.tests]
+        return test_status.count(TestResult.WARN)
+
+    @property
+    def errors(self):
```

Comment:
Nitpick - to be similar to the other properties` naming convention
```suggestion
    def errored(self):
```

Reply:
It would be nice, but that's how avocado defines it.